packages/autorest.typescript/src/generators/clientFileGenerator.ts

```diff
@@ -142,7 +144,7 @@ export function generateClient(clientDetails: ClientDetails, project: Project) {
     });
     clientFile.addImportDeclaration({
       namedImports: ["createLroSpec"],
-      moduleSpecifier: `./lroImpl`
+      moduleSpecifier: getImportModuleName(`./lroImpl`, moduleKind)
```

Comment:
Introduced some convenience for file import (as opposed to a directory import) where you can just pass the cjs name and it'll append `.js` if moduleKind is esm